opm/autodiff/BlackoilPropsAdFromDeck.cpp

```diff
@@ -141,18 +146,41 @@ namespace Opm
                 if (numSamples > 0) {
                     auto splinePvt = std::shared_ptr<PvtDeadSpline>(new PvtDeadSpline);
                     splinePvt->initFromOil(pvdoTables, numSamples);
+
+                    if (!eclState->getOilvisctTables().empty()) {
+                        splinePvt->setOilvisctTables(eclState->getOilvisctTables(),
+                                                     deck->getKeyword("VISCREF"));
+                    }
+
                     props_[phase_usage_.phase_pos[Liquid]] = splinePvt;
                 } else {
                     auto deadPvt = std::shared_ptr<PvtDead>(new PvtDead);
                     deadPvt->initFromOil(pvdoTables);
+
+                    if (!eclState->getOilvisctTables().empty()) {
+                        deadPvt->setOilvisctTables(eclState->getOilvisctTables(),
+                                                   deck->getKeyword("VISCREF"));
+                    }
+
                     props_[phase_usage_.phase_pos[Liquid]] = deadPvt;
                 }
             } else if (!pvtoTables.empty()) {
-                props_[phase_usage_.phase_pos[Liquid]].reset(new PvtLiveOil(pvtoTables));
```

Comment:
ok, changed. note that the pointer is still called `splinePvt` or `deadPvt` for the PVDO cases...